enterprise/frontend/src/metabase-enterprise/clean_up/index.tsx

```diff
@@ -1,34 +1,37 @@
 import { t } from "ttag";
 
+import {
+  isInstanceAnalyticsCustomCollection,
+  isTrashedCollection,
+} from "metabase/collections/utils";
 import { ModalRoute } from "metabase/hoc/ModalRoute";
+import * as Urls from "metabase/lib/urls";
 import { PLUGIN_COLLECTIONS } from "metabase/plugins";
 import { hasPremiumFeature } from "metabase-enterprise/settings";
 
 import { CleanupCollectionModal } from "./CleanupCollectionModal";
 import { CollectionCleanupAlert } from "./CollectionCleanupAlert";
 
 if (hasPremiumFeature("collection_cleanup")) {
-  PLUGIN_COLLECTIONS.canCleanUp = true;
-
-  PLUGIN_COLLECTIONS.getCleanUpMenuItems = (
-    itemCount,
-    url,
-    isInstanceAnalyticsCustom,
-    isTrashed,
-    canWrite,
-  ) => {
-    const canCleanUpCollection =
-      itemCount !== 0 && !isInstanceAnalyticsCustom && !isTrashed && canWrite;
+  PLUGIN_COLLECTIONS.canCleanUp = collection => {
+    return Boolean(
+      !isInstanceAnalyticsCustomCollection(collection) &&
+        !isTrashedCollection(collection) &&
+        !collection.is_sample &&
+        collection.can_write,
+    );
+  };
```

Comment:
can this be a real function with a few unit tests? 🙏

Reply:
done-zos